core/src/test/java/org/apache/iceberg/io/TestResolvingIO.java

```diff
@@ -51,16 +61,17 @@ public void testResolvingFileIOWithHadoopFileIOKryoSerialization() throws IOExce
     resolvingFileIO.setConf(conf);
     resolvingFileIO.initialize(ImmutableMap.of("k1", "v1"));
 
-    assertThat(resolvingFileIO.ioClass(temp.toString())).isEqualTo(HadoopFileIO.class);
-    assertThat(resolvingFileIO.newInputFile(temp.toString())).isNotNull();
+    assertThat(resolvingFileIO.ioClass(tempDir.getCanonicalPath())).isEqualTo(HadoopFileIO.class);
```

Comment:
I might have missed the discussion but could you elaborate please why we switched `temp` from `java.nio.file.Path` to `File` and why this change here is necessary?

Reply:
I think we should revert those changes and rather just define `@TempDir private java.nio.file.Path temp`, since it seems the main reason is most likely to not have a clash with Hadoop's Path class

Reply:
thank you @nastra, it was coming from my attempt to resolve conflict and desired to simplify the tempDir usage in unit tests. Now i replaced `tempDir` in favor of `@TempDir private java.nio.file.Path temp` per your suggestion